autogpt_platform/backend/backend/blocks/github/_api.py

```diff
@@ -0,0 +1,50 @@
+from urllib.parse import urlparse
+
+from backend.blocks.github._auth import GithubCredentials
+from backend.util.request import Requests
+
+
+def _validate_github_url(url: str) -> None:
+    parsed_url = urlparse(url)
+    if parsed_url.netloc != "github.com":
```

Comment:
github can also be hosted via enterprise on custom urls ex: github.agpt.co so this probably isn't a good design choice

Reply:
imo its insanity we require urls at all rather than the repo, owner, and source and then just build from there

Reply:
this is a series of nits ^

Reply:
The two nits are contradictory. 

I removed the github.com check validation entirely, since we are replacing everything to api.github.com

> github can also be hosted via enterprise on custom URLs
so maybe you need to provide an option either way for changing the host because we are ignoring them.
The existing implementation doesn't support this. I don't even know if the self-hosted github would also have an API or not.